src/utils.h

```diff
@@ -28,3 +32,30 @@ limitations under the License.
                   << std::endl;                                                \
         abort();                                                               \
     } while (false)
+
+// According to the agreement, Random Value Generator is the only way to get any random value in YARPGen. It is used for different random decisions all over the source code.
+class RandValGen {
+  public:
+    // Specific seed can be passed to constructor to reproduce the test.
+    // Zero value is reserved (it notifies RandValGen that it can choose any)
+    RandValGen(uint64_t _seed);
+
+    template <typename T> T getRandValue(T from, T to) {
```

Comment:
Should we have a version, which doesn't take any parameters, but returns the value in the whole type range?

Reply:
Maybe. We also need to think about a version that accepts IRValue.

Reply:
Make sense, you can add TODO for that.

Reply:
Done